CRM/Event/Form/Participant.php

```diff
@@ -1611,7 +1603,7 @@ protected function addParticipant($params, $contactID) {
       );
       $participantParams['id'] = $pID;
     }
-    $participantParams['discount_id'] = CRM_Core_BAO_Discount::findSet($this->getEventID(), 'civicrm_event') ?: 'null';
+    $participantParams['discount_id'] = $this->getSubmittedValue('discount_id');
```

Comment:
This shouldn't fall back to the lookup - cos whatever was selected on the form should be respected